### ISSUE TITLE
Add parent_id in sendStartTypingEvent()

### ISSUE THREAD

**Issue by EuyjaeHwang** [GetStream/stream-chat-swift#1828]
## What are you trying to achieve?
In [Typing Indicators](https://getstream.io/chat/docs/ios-swift/typing_indicators/?language=swift) docs says 
>Use parent_id field of the event to indicate that typing is happening in a thread

But I can't find the way how to add `parent_id` in `channelController.sendStartTypingEvent()`.
Also,  I wanna distinguish who is typing in a thread. 
However, SDK just returns set of ChatUser without parent message id

## What would be the better way?

just add parent message id to current APIs like following examples

```Swift
func sendStartTypingEvent(parentMessageId: String?, completion: ((Error?) -> Void)? = nil) {}
func sendStopTypingEvent(parentMessageId: String?, completion: ((Error?) -> Void)? = nil) {}

func channelController(
    _ channelController: ChatChannelController,
    didChangeTypingUserEvent event: Set<TypingUserEvent>
) {
  ...
}

struct TypingUserEvent {
  var users: Set<ChatUser>
  var parentMessageId: String?
}
```

Sincerely.

## GetStream Environment
**GetStream Chat version:** 4.10.1
**GetStream Chat frameworks:** StreamChat
**iOS version:** 15.x, 14.x
**Swift version:** 5.5.2
**Xcode version:** 13.2.1
**Device:** all of iPhone, iPad supporting iOS 14+

**Comment by adamrushy** [comment]
Hello, @EuyjaeHwang thanks for the report.

We have this ticket in our backlog of work and will bump the priority to get this included within the iOS SDK. Thanks again, and I will update you as soon as we start working on it.

Thanks,
Adam

**Comment by polqf** [comment]
Hello! We just released 4.12.0, you can check the changelog [here](https://github.com/GetStream/stream-chat-swift/releases/tag/4.12.0).
This issue should be fixed now. Feel free to reopen it if needed

**Comment by polqf** [comment]
Hi @EuyjaeHwang, we just released [4.17.0](https://github.com/GetStream/stream-chat-swift/releases/tag/4.17.0) which contains a fix for the issue you reported.

Let us know if we can help you with something else
